backend/app/api/endpoints/knowledge.py

```
@@ -1629,140 +1601,709 @@ def _update_kb_summary_after_deletion(kb_id: int, user_id: int, user_name: str):
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=str(e),
+        )
+
+
+@permission_request_router.delete(
+    "/{request_id}", response_model=PermissionRequestResponse
+)
+def cancel_permission_request(
+    request_id: int,
+    current_user: User = Depends(security.get_current_user),
```

Ruff (0.14.14)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)